frontend/app/src/scenes/Home/scenes/AddSubCategory/index.tsx

```diff
@@ -14,26 +14,36 @@ import {
   Typography,
 } from 'antd';
 import mainStyles from '../../styles.css';
-import axios, { AxiosResponse } from 'axios';
+import axios, { AxiosResponse, Method } from 'axios';
 import { handleApiError } from '../../../../services/util/errorHandler';
 
 const { Title, Text } = Typography;
 const { Option } = Select;
 
 class AddSubCategory extends React.Component<
-  RouteComponentProps,
+  RouteComponentProps<SubCategoryUrlParams>,
   AddSubCategoryStateProps
 > {
-  constructor(props: RouteComponentProps) {
+  subCategoryId: string | null;
+  constructor(props: RouteComponentProps<SubCategoryUrlParams>) {
     super(props);
+    this.subCategoryId = this.props.match.params.subCategoryId;
     this.state = {
       isLoading: false,
       categories: [],
+      subcategory: null,
+      isComponentEdit: false,
```

Comment:
instead of saving a boolean. Store the type which accepts only "add" and "edit"

Reply:
Also, it shouldn't be a state variable since you don't need to rerender the component when the component type changes. (the type isn't changing time to time)

Reply:
yes I will change it.